components/script/dom/bindings/typedarrays.rs

```diff
@@ -142,3 +160,25 @@ where
         TypedArray::from(dest.get())
     }
 }
+
+pub fn create_new_external_array_buffer(
+    cx: JSContext,
+    mapping: Rc<RefCell<Vec<u8>>>,
```

Comment:
lets turn the original mapping into an `Arc<Mutex<Vec<u8>>>`, because [`free_func` must be threadsafe](https://github.com/servo/mozjs/blob/1d0ba1dc0c40008cfea6f14271844558dbea3cbd/mozjs-sys/mozjs/js/public/ArrayBuffer.h#L89)

Reply:
Thanks, converted it to Arc<Mutex> can you check it again?